### ISSUE TITLE
Big (N-nary) operators are at the wrong vertical alignment

### ISSUE THREAD

**Issue by ronkok** [stipub/stixfonts#268]
For big (N-nary) operators, STIX TWO, like other math fonts, has both a standard and an alternate glyph. The alternate glyph is used in what TeX calls `displaystyle` and what MathML calls `display="block"`. 

Some of the alternate glyphs in STIX TWO are at the wrong vertical alignment. They are placed entirely above the baseline. They should instead have a substantial depth (descender) below the baseline. The image below shows the glyphs for Unicode points U+22C2 and U+22C3.

![STIX Issue](https://github.com/stipub/stixfonts/assets/16403058/5692db96-45b0-4d86-b3f0-5b6af7c98959)

Latin Modern gets it right. STIX TWO gets it wrong. The results are appearing in production. The Temml library has an [open issue](https://github.com/ronkok/Temml/issues/66) that is the direct result of this misalignment.

**Comment by tiroj** [comment]
Noted for fix.

**Comment by ronkok** [comment]
I've done some more investigation. Below is a list of characters that all have this problem.

| LaTeX function   | Character | Code Point |
|------------------|:---------:|:----------:|
| \\bigwedge       | ⋀         | U+22C0     |
| \\bigvee         | ⋁         | U+22C1     |
| \\bigcap         | ⋂         | U+22C2     |
| \\bigcup         | ⋃         | U+22C3     |
| \\bigodot        | ⨀         | U+2A00     |
| \\bigoplus       | ⨁         | U+2A01     |
| \\bigotimes      | ⨂         | U+2A02     |
| \\bigcupdot      | ⨃         | U+2A03     |
| \\biguplus       | ⨄         | U+2A04     |
| \\bigsqcap       | ⨅         | U+2A05     |
| \\bigsqcup       | ⨆         | U+2A06     |
| \\bigdoublevee   | ⨇         | U+2A07     |
| \\bigdoublewedge | ⨈         | U+2A08     |
| \\bigtimes       | ⨉         | U+2A09     |

The integration symbols and the summation symbol all look good.

**Comment by Enivex** [comment]
Note that in Latex operators like this are always centered vertically regardless of how they appear in the font, so that's why this doesn't have any effect there.

Temml should probably do the same if possible?

**Comment by ronkok** [comment]
> Note that in Latex (and other places) operators like this are always centered vertically

I believe the same should occur in MathML. It does occur in Firefox. It's broken in Chromium and WebKit. It's a bug. Google has been very unresponsive to other MathML bugs that I have reported. I was hoping you might be able to help me work around this issue and get a good rendering even if they don't respond.

> you can set the symmetric property?

Yes, I can set the symmetric attribute for an `<mo>` element, although it should not be necessary. These character code points are in the MathML operator dictionary, which states that they should have the `symmetric`, `largeop`, and `movablelimits` pre-set to `true`. The alignment is still wrong. I have also done experiments with setting those attributes manually. It does not help.

This bug may have its origin in the MathML-Core specification. When I read section [3.2.4.5 Layout of operators](https://www.w3.org/TR/mathml-core/#layout-of-operators), I note that part 2 is conditioned on the `stretchy` attribute and specifies an adjustment of the vertical alignment for `stretchy` elements. Part 3, regarding the `largeop` attribute is silent regarding vertical alignment.

Indeed, I can achieve the correct vertical alignment in Chromium and WebKit by setting the `stretchy` attribute. But I then get the small glyph, not the large glyph. That's true even when I manually set the `largeop` attribute. The end result is that I can get the correct glyph or the correct vertical alignment but not both.

I intend to take the following steps:

1. Open issues with the MathML-Core spec and with both browsers. I do not expect much help from any of them.
2. Try for a CSS fix that enforces the correct glyph while `stretchy` is set. It would help if you could tell me the `font-feature-setting` that would select the large glyph. `aalt`?, `ss01`? other?
3. If I cannot enforce the correct glyph, I guess I'll use CSS to manually change the vertical alignment. That will cause a breaking change someday when the browsers finally get this right. I'd like to avoid that, but I may have no choice.

**Comment by ronkok** [comment]
Strike option 3 above. I have discovered that I can get proper rendering if the `<munderover>` element has the `stretchy` attribute and is a top-level element. That is, it must not be nested within a `<mrow>` element.

I consider that to be very odd behavior and I may go mad someday working around browser vagaries but at least I have a path forward.

**Comment by tiroj** [comment]
Questions:

Would it break anything in the implementations that get the positioning right if I were to lower these glyphs within their glyph space (obvously I might need to update some MATH table data for these also)?

Would it help anywhere?

**Comment by ronkok** [comment]
> Would it break anything in the implementations that get the positioning right ... ?

No, provided that you also make the appropriate changes in the MATH table. Firefox already gets vertical alignment correct for the Latin Modern font and that font already has glyphs positioned as you now propose to do in STIX.

BTW, The MathML-Core committee is moving toward a correction to the spec.  See [the issue](https://github.com/w3c/mathml-core/issues/250) and [the PR](https://github.com/w3c/mathml-core/pull/276). There was a committee meeting on Sundary which discussed this issue among others. I don't know what the committee decided.

Of course, a correction to the spec is just one step. After that we would also need action taken by the browser vendors and they are very slow. So your proposal would definitely help.

**Comment by tiroj** [comment]
All the big forms of characters noted in https://github.com/stipub/stixfonts/issues/268#issuecomment-2179212545 have been updated for v2.25. In addition to being lowered, they have also been adjusted so that their size matches that of the big integration and summation signs, and some of the regular size glyphs were also adjusted slightly.

<img width="607" alt="Image" src="https://github.com/user-attachments/assets/040aef1c-69e9-4066-a479-09c0d9443597" />

**Comment by ronkok** [comment]
This is great news. Thank you.

**Comment by ronkok** [comment]
Follow up: I am updating the [Temml](https://temml.org/) library today and I can confirm that I am getting excellent rendering of all these characters when using STIX TWO v2.14. The rendering is now excellent in Chromium, Firefox, and WebKit. I'll have a new Temml release out tomorrow.

Thank you for the great work.

**Comment by Enivex** [comment]
> Follow up: I am updating the [Temml](https://temml.org/) library today and I can confirm that I am getting excellent rendering of all these characters when using STIX TWO v2.14. The rendering is now excellent in Chromium, Firefox, and WebKit. I'll have a new Temml release out tomorrow.
> 
> Thank you for the great work.

I thought these changes weren't available yet, or were they backported?

**Comment by tiroj** [comment]
The rendering of these forms is environment-dependent. So the v2.14 glyphs, although sitting too high in the font, may be still be positoned correctly within equations.

In v2.25, I have lowered them in the font, but that also means that the change needs to be tested in various environments, so I am reopening this issue until v2.25 ships and we are able to review.
